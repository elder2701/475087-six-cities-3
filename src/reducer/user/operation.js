import {ActionCreator, AuthorizationStatus} from "./user.js";

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
        .then((res) => {
          dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
          dispatch(ActionCreator.loadUserInfo(res.data));
        })
        .catch((err) => {
          throw err;
        });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
    .then((res) => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.loadUserInfo(res.data));
    });
  },
};

export {Operation};
