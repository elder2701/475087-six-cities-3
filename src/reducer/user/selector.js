import NameSpace from "../name-space.js";

const getAuthStatus = (state)=> state[NameSpace.USER].authorizationStatus;

export {getAuthStatus};
