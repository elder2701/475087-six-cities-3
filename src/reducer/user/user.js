import {extend} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.AUTH,
  userInfo: {}
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  LOAD_USER_INFO: `LOAD_USER_INFO`
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  loadUserInfo: (userInfo) => {
    return {
      type: ActionType.LOAD_USER_INFO,
      payload: userInfo
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.LOAD_USER_INFO:
      return extend(state, {
        userInfo: action.payload
      });
  }
  return state;
};

export {ActionCreator, reducer, AuthorizationStatus};
