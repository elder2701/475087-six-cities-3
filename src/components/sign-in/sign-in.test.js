import React from "react";
import renderer from "react-test-renderer";
import SignIn from "./sign-in.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {Router} from "react-router-dom";
import NameSpace from "../../reducer/name-space.js";
import history from "../../history";

const mockStore = configureStore([]);

it(`<SignIn /> sould be render`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offers: {
        Amsterdam: {
          city: {
            name: `Amsterdam`,
            location: {
              latitude: 48.85661,
              longitude: 2.351499,
              zoom: 13
            }
          },
          offers: [
            {
              images: [``, ``],
              title: `Amazing and Extremely Central Flat`,
              rating: 1,
              type: `apartment`,
              bedrooms: 4,
              price: 467,
              goods: [`Breakfast`, `Laptop friendly workspace`],
              description: ``,
              location: {
                latitude: 4,
                longitude: 2,
                zoom: 16
              },
              id: 1,
              previewImage: ``,
              isFavorite: false,
              isPremium: false,
              maxAdults: 5,
              hostId: 25,
              hostName: `Angelina`,
              hostIsPro: true,
              hostAvatarUrl: ``
            },
            {
              images: [``],
              title: `Waterfront with extraordinary view`,
              rating: 3,
              type: `room`,
              bedrooms: 1,
              price: 133,
              goods: [`Washer`, `Laptop friendly workspace`],
              description: ``,
              location: {
                latitude: 78.85761,
                longitude: 2.358499,
                zoom: 16
              },
              id: 2,
              previewImage: ``,
              isFavorite: false,
              isPremium: false,
              maxAdults: 2,
              hostId: 25,
              hostName: `Angelina`,
              hostIsPro: true,
              hostAvatarUrl: ``
            }
          ]
        }
      },
      Paris: {
        city: {
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        },
        offers: []
      }
    },
    [NameSpace.CITY]: {city: `Amsterdam`},
    [NameSpace.USER]: {authorizationStatus: AuthorizationStatus.AUTH}
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <SignIn submit={()=>{}} />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
