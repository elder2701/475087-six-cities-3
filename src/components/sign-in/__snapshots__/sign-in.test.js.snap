// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> sould be render 1`] = `
<main
  className="page__main page__main--login"
/>
`;
